tpi/terraform.py

```diff
@@ -42,29 +33,26 @@ def create(self, name: Optional[str] = None, **config):
         from tpi import render_json
 
         assert name and "cloud" in config
-        with self.make_tf(name) as tf:
-            tf_file = os.path.join(tf.working_dir, "main.tf.json")
-            with open(tf_file, "w", encoding="utf-8") as fobj:
-                fobj.write(render_json(name=name, **config, indent=2))
-            tf.cmd("init")
-            tf.cmd("apply", auto_approve=IsFlagged)
+        tf_file = os.path.join(self.working_dir, "main.tf.json")
+        with open(tf_file, "w", encoding="utf-8") as fobj:
+            fobj.write(render_json(name=name, **config, indent=2))
+        self.tf.cmd("init")
+        self.tf.cmd("apply", auto_approve=IsFlagged)
```

Comment:
The one thing to note now is that since the __init__'ed `working_dir` is always used (instead of separate dirs per resource `name`), this class is currently only useful for managing a single `iterative-machine` resource at a time (since we just rewrite the entire `main.tf.json` any time a config change is made).

A proper TPI release should probably parse the entire existing `main.tf.json` and only add/modify/delete the appropriate named resource entries so that you can actually manage multiple resources at a time.